spec/test_spec.rb

```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /index' do
+    before(:example) { get '/users' }
+
+    it 'should returns http status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'should render the index template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'should render the correct text in the template' do
+      expect(response.body).to include('Here is a list of posts for a given user')
+    end
+  end
+
+  describe 'GET /show' do
+    before(:example) { get '/users/1' }
+
+    it 'should return correct response' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'should render the show template ' do
+      expect(response).to render_template('show')
+    end
+
+    it 'should have the correct text in the template' do
+      expect(response.body).to include('Profile of')
+    end
+  end
+end
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET /index' do
+    before(:example) { get '/users/1/posts' }
+
+    it 'should returns http status' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'should render the index template' do
+      expect(response).to render_template('index')
+    end
+
+    it 'should render the correct text in the template' do
+      expect(response.body).to include('Post interactions for a given user ')
+    end
+  end
+
+  describe 'GET /show' do
+    before(:example) { get '/users/1/posts/1' }
+
+    it 'should return correct response' do
+      expect(response).to have_http_status(:ok)
+    end
+
+    it 'should render the show template ' do
+      expect(response).to render_template('show')
+    end
+
+    it 'should have the correct text in the template' do
+      expect(response.body).to include('Posts')
+    end
```

Comment:
- [ ] I would recommend making another file for post spec as `post_spec.rb` under `spec/requests/` folder.

- [ ] Also name this file as `users_spec.rb` and put under `spec/requests/` folder. It's better if we put each controller test in a separate folder so that one can easily check specific controller specs if the project gets bigger.

Reply:
updated required changes